### PR TITLE
Implement installation process

### PR DESCRIPTION
**Pull request by jerrymarino**
Normalize installs to a canonical location
Update the plist to unset the env var if necessary

### CODE REVIEW

File: utils/install_build_service_app.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+mkdir -p "$INSTALL_DIR"
+EXISTING_BUILD_SERVICE="$(find "$INSTALL_DIR" -name *.app | head)"
+
+function get_version() {
+    /usr/libexec/PlistBuddy "$1/Contents/Info.plist" -c "Print :CFBundleVersion"
+}
+
+# Print a message when replacing
+if [[ -d "$EXISTING_BUILD_SERVICE" ]]; then
```
Comment (rahul-malik) [inline utils/install_build_service_app.sh]: This script is fine for now but should we just create a installer `pkg` and let macOS perform this kind of installation?

Comment (jerrymarino) [reply utils/install_build_service_app.sh]: How would the rest of the install process work with `pkg`?
- code generate a plist with the binary path ( conditionally select to ~/Library or /LaunchDaemons for global )
- call launchctl
- ask the user to close xcode and reopen

**Comment by jerrymarino** [comment]
@rahul-malik thanks for the review, going to land this for now so we can continue to test it. I'll look to see what streamline install with `pkgutil` as a followup. I don't know if it's possible to do all of the things currently in there, but I think for releases' it'd sure be nice to distribute as an installer package

**Comment by jerrymarino** [comment]
@rahul-malik looks like we can use a `postinstall` script in a `.pkg` to achieve all the dynamic behavior. This has the added benefit that we can launch the installer process as an external process.

I couldn't find any bazel rules for this so I wrote one. We might consider upstreaming to `rules_apple` or `rules_pkg`

**Pull request merged**